cl/search/management/commands/pacer_bulk_fetch.py

```diff
@@ -0,0 +1,385 @@
+            type=str,
+            choices=["fetch", "process"],
+            default="fetch",
+            help="Stage of the command to run: fetch or process",
+        )
+
+    @staticmethod
+    def docs_to_process_cache_key():
+        """Helper method to improve testability."""
+        return "pacer_bulk_fetch.docs_to_process"
+
+    @staticmethod
+    def timed_out_docs_cache_key():
+        """Helper method to improve testability."""
+        return "pacer_bulk_fetch.timed_out_docs"
+
+    @staticmethod
+    def setup_logging(testing: bool = False) -> None:
+        if not testing:
+            logging.basicConfig(
+                filename=f'pacer_bulk_fetch_{datetime.now().strftime("%Y%m%d_%H%M%S")}.log',
+                level=logging.INFO,
+                format="%(asctime)s - %(levelname)s - %(message)s",
+            )
+
+    def setup_celery(self) -> None:
+        """Setup Celery by setting the queue_name and throttle."""
+        self.queue_name = self.options.get("queue_name", "pacer_bulk_fetch")
+        self.throttle = CeleryThrottle(queue_name=self.queue_name)
+
+    def handle_pacer_session(self) -> None:
+        """Make sure we have an active PACER session for the user."""
+        self.pacer_username = self.options.get(
+            "pacer_username", settings.PACER_USERNAME
+        )
+        self.pacer_password = self.options.get(
+            "pacer_password", settings.PACER_PASSWORD
+        )
+        get_or_cache_pacer_cookies(
+            self.user.pk,
+            username=self.pacer_username,
+            password=self.pacer_password,
+        )
+
+    def set_user(self, username: str) -> None:
+        """Get user or raise CommandError"""
+        if not username:
+            raise CommandError(
+                "No username provided, cannot create PacerFetchQueues."
+            )
+        try:
+            self.user = User.objects.get(username=username)
+        except User.DoesNotExist:
+            raise CommandError(f"User {username} does not exist")
+
+    def identify_documents(self) -> None:
+        """Get eligible documents grouped by court"""
+        filters = [
+            Q(pacer_doc_id__isnull=False),
+            Q(is_available=False),
+        ]
+        if self.options.get("min_page_count"):
+            filters.append(Q(page_count__gte=self.options["min_page_count"]))
+        if self.options.get("max_page_count"):
+            filters.append(Q(page_count__lte=self.options["max_page_count"]))
+
+        # Do not attempt to fetch docs that were already fetched:
+        cached_fetches = cache.get(self.docs_to_process_cache_key(), [])
+        previously_fetched = [rd_pk for (rd_pk, _) in cached_fetches]
+        # Only try again with those that were timed out before:
+        cached_timed_out = cache.get(self.timed_out_docs_cache_key(), [])
+        previously_timed_out = [rd_pk for (rd_pk, _) in cached_timed_out]
+        redundant = set(previously_fetched) - set(previously_timed_out)
```

Comment:
Perhaps `ids_to_skip` is more clear? 

```suggestion
        ids_to_skip = set(previously_fetched) - set(previously_timed_out)
```

Reply:
changed